jdaviz/app.py

```diff
@@ -1069,7 +1069,55 @@ def get_viewer_ids(self, prefix=None):
     def get_viewer_reference_names(self):
         """Return a list of available viewer reference names."""
         # Cannot sort because of None
-        return [self._viewer_item_by_id(vid)['reference'] for vid in self._viewer_store]
+        return [self._viewer_item_by_id(vid).get('reference') for vid in self._viewer_store]
+
+    def get_first_viewer_reference_name(
+            self, require_no_selected_data=False,
+            require_spectrum_viewer=False,
+            require_spectrum_2d_viewer=False,
+            require_table_viewer=False,
+            require_flux_viewer=False,
+            require_image_viewer=False
+    ):
+        """
+        Return the viewer reference name of the first available viewer.
+        Optionally use ``require_no_selected_data`` to require that the
+        viewer has not yet loaded data, or e.g. ``require_spectrum_viewer``
+        to require that the viewer supports spectrum visualization.
+        """
+        from jdaviz.configs.specviz.plugins.viewers import SpecvizProfileView
+        from jdaviz.configs.specviz2d.plugins import SpectralExtraction
+        from jdaviz.configs.cubeviz.plugins.viewers import CubevizProfileView, CubevizImageView
+        from jdaviz.configs.mosviz.plugins.viewers import (
+            MosvizProfileView, MosvizTableViewer, MosvizProfile2DView
+        )
```

Comment:
What are best practices for "inner" imports like this? I've always avoided since I've been unaware. I summon the arbiter @pllim!

Reply:
Ooops, just saw this. This delays the import until `_get_first_viewer_reference_name` is called, but I think that is called pretty soon anyway, so 🤷

Reply:
Actually, I think this mostly fixed non-GUI mode performance issues like #881 and #1362 ❗